x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/EnrichPolicy.java

```diff
@@ -68,7 +70,24 @@ private static void declareParserOptions(ConstructingObjectParser<?, ?> parser)
     }
 
     public static EnrichPolicy fromXContent(XContentParser parser) throws IOException {
-        return PARSER.parse(parser, null);
+        Token token = parser.currentToken();
```

Comment:
Not super happy with this parsing code, but in other to use object parser, another class would need to be introduced for policy types and since there is currently only policy type, this seems overkill to me. I think the important thing here is that the policy format is future proof now.